pinot-core/src/main/java/org/apache/pinot/core/io/writer/impl/MutableOffHeapByteArrayStore.java

```diff
@@ -157,7 +150,7 @@ public void close()
       _pinotDataBuffer.close();
     }
 
-    private long getSize() {
+    private int getSize() {
```

Comment:
I think we should support long. At the time this was coded, Lbuffer support had bugs.

Reply:
The total size could be long, while each buffer's size is limited to 2G.
Besides, for real-time forward index/dictionary, I don't think we should allow buffers of gigabytes because that will definitely cause problem when committing the segment.